src/Workspaces/Core/Portable/SolutionCrawler/BackgroundAnalysisScope.cs

```diff
@@ -6,11 +6,32 @@ namespace Microsoft.CodeAnalysis.SolutionCrawler
 {
     internal enum BackgroundAnalysisScope
     {
-        ActiveFile,
-        OpenFilesAndProjects,
-        FullSolution,
+        // NOTE: Do not change existing field ordering/values as this scope is saved as a user option,
+        //       and that would break users who have saved a non-default scope for the option.
 
-        Minimal = ActiveFile,
-        Default = OpenFilesAndProjects,
+        /// <summary>
+        /// Analyzers are executed only for currently active document.
+        /// Compiler analyzer is treated specially and executed for all open documents.
+        /// </summary>
+        ActiveFile = 0,
+
+        /// <summary>
+        /// All analyzers, including compiler analyzer, are executed for all open documents.
+        /// </summary>
+        OpenFiles = 1,
+
+        /// <summary>
+        /// All analyzers, including compiler analyzer, are executed for all documents in the current solution.
+        /// </summary>
+        FullSolution = 2,
+
+        /// <summary>
+        /// Analyzers are disabled for all documents.
+        /// Compiler analyzer is treated specially and executed for all open documents.
```

Comment:
yup.  this would be better handled as two internal enums IMO.

Reply:
Lets discuss if we just want to add a separate user facing bool option (checkbox) to determine whether or not to show compiler diagnostics for closed files.

Reply:
note, what i'm saying is:
*regardless* of the user facing side of this, it would be cleaner IMO if we represented this *internally* as two enums (or an enum+bool, whatever).

Those two values would end up being used to populate the normal list of options that you have in your PR.  It's just that we wouldn't be trying to haev a single enum internall with this funky behavior.

--

Note: anotehr approach would be to have this be a single enum, just with 'Flags' behavior

Reply:
> it would be cleaner IMO if we represented this internally as two enums (or an enum+bool, whatever)

@CyrusNajmabadi I have updated the doc comments on this file to remove mention of compiler analyzer. Note that we have just one place in IDE diagnostic service where we special case compiler analyzer to decide whether or not it should run on a given document. We don't need to add a separate enum or bool in our system for this, we just have a descriptive comment at the single place where we have this deduction logic from BackgroundAnalysisScope option value.